src/app/data-planes/data/stats.ts

```diff
@@ -33,8 +42,44 @@ export const parse = (lines: string): EnvoyStats => {
       }
     })(value.join(':').trim())
 
+    let _key = key
+
+    // if we are a `listener.` prefixed line we need to avoid ipv4 and ipv6
+    // addresses. This turns the line into something similar to the rest i.e.
+    // listener.10.244.0.11_8081.http.edge-gateway.downstream_rq_1xx to
+    // listener.http.edge-gateway_8081.downstream_rq_1xx
+    if (key.startsWith('listener.')) {
+      if (!['.http.', '.tcp.'].some(item => key.includes(item))) {
+        // if its not a http or tcp one, ignore for now
+        return prev
+      }
+
+      // get rid of listener.
+      let parts = key.split('.').slice(1)
+      // save the _key to search on later
+      _key = parts.join('.')
+
+      // get rid of the actual end key name so we don't get a false _1xx
+      parts.pop()
+
+      // search for a port in the whats left
+      const pos = parts.join('.').search(trailingPortRe)
+      if (pos === -1) {
+        // if we didn't find a port, ignore
+        return prev
+      }
+
+      // 10.244.0.11_8081.http.edge-gateway.downstream_rq_1xx
+      parts = _key.substring(pos + 1).split('.')
+
+      // 8081.http.edge-gateway.downstream_rq_1xx
+      const [port, protocol, service, name] = parts
+      parts = ['listener', protocol, service.search(trailingPortRe) !== -1 ? service : `${service}_${port}`, name]
+      // listener.http.edge-gateway_8081.downstream_rq_1xx
+      _key = parts.join('.')
+    }
```

Comment:
We are eventually gonna go back to not using `listener.` at all, which I'm glad about 😅 , but we need some changes made so its possible to get the non-rolled up info from the `/stats` endpoint. I won't be spending too much time tweaking this code whatever as it will go pretty soon.